### PR TITLE
Add cluster-autoscaler version for Kubernetes 1.24

### PR DESCRIPTION
**Pull request by embik**
**What does this PR do / Why do we need it**:
A couple of components do not have Kubernetes 1.24 versions. cluster-autoscaler 1.24.0 just recently released. This PR adds it.

In addition, I found a seemingly ancient `clusterautoscaler` package that apparently last worked with Kubernetes **1.14**. I took the liberty to drop it from our code base as part of this PR. 

**Does this PR close any issues?**:
towards #9818

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/seed-controller-manager/kubernetes/resources.go [hunk 1]
```diff
@@ -325,9 +324,6 @@ func GetDeploymentCreators(data *resources.TemplateData, enableAPIserverOIDCAuth
 		)
 	}
 
-	if data.Cluster().Annotations[kubermaticv1.AnnotationNameClusterAutoscalerEnabled] != "" {
```
Comment (xrstf) [inline pkg/controller/seed-controller-manager/kubernetes/resources.go:328]: Does the constant then still need to exist?

Comment (embik) [reply pkg/controller/seed-controller-manager/kubernetes/resources.go:328]: I don't think so, good point. Let me remove it.

**Review by xrstf — APPROVED** [review]
/approve

**Comment by xrstf** [comment]
This accidentally also fixed #7069.

**Pull request merged**